asserts/snap_asserts.go

```diff
@@ -698,3 +698,121 @@ func InitBuiltinBaseDeclaration(headers []byte) error {
 	builtinBaseDeclaration = a.(*BaseDeclaration)
 	return nil
 }
+
+// TODO(matt): consider renaming this ... Developer is a bit vague.
+type Developer struct {
+	AccountID string
+	Since     time.Time
+	Until     time.Time
+}
+
+// SnapDeveloper holds a snap-developer assertion, defining the developers who
+// can collaborate on a snap while it's owned by a specific publisher.
+type SnapDeveloper struct {
+	assertionBase
+	developers []*Developer
+}
+
+// SnapID returns the snap id of the snap.
+func (snapdev *SnapDeveloper) SnapID() string {
+	return snapdev.HeaderString("snap-id")
+}
+
+// PublisherID returns the publisher's account id.
+func (snapdev *SnapDeveloper) PublisherID() string {
+	return snapdev.HeaderString("publisher-id")
+}
+
+// Developers returns the developers allowed to collaborate on the snap.
+func (snapdev *SnapDeveloper) Developers() []*Developer {
+	return snapdev.developers
+}
+
+func (snapdev *SnapDeveloper) checkConsistency(db RODatabase, acck *AccountKey) error {
+	// Check authority is the publisher or trusted.
+	authorityID := snapdev.AuthorityID()
+	publisherID := snapdev.PublisherID()
+	if !db.IsTrustedAccount(authorityID) && (publisherID != authorityID) {
+		return fmt.Errorf("snap-developer must be signed by the publisher or a trusted authority: authority-id=%q, publisher-id=%q", authorityID, publisherID)
+	}
+
+	_, err := db.Find(SnapDeclarationType, map[string]string{
+		// XXX: mediate getting current series through some context object? this gets the job done for now
+		"series":  release.Series,
+		"snap-id": snapdev.SnapID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("snap-developer assertion for snap id %q does not have a matching snap-declaration assertion", snapdev.SnapID())
+	}
+
+	// TODO(matt): check publisher-id exists (only needed if different to account-id")
+	// TODO(matt): check that snap-decl's publisher is snap-dev's publisher? not sure about this.
```

Comment:
No, we don't want this I think. Part of the benefit of having the publisher here is precisely to allow us to update the publisher without having to do both snap-declaration and snap-developer updates atomically, which means we must be allowed to do developer first before the declaration reflects it.